vip-jetpack/connection-pilot/class-jetpack-connection-controls.php

```diff
@@ -138,12 +138,20 @@ public static function connect_site( bool $skip_connection_tests = false, bool $
 	 * @return bool True if connection worked, false otherwise
 	 */
 	public static function connect_akismet(): bool {
-		if ( class_exists( 'Akismet_Admin' ) ) {
+		$current_user = wp_get_current_user();
+		$vip_user = get_user_by( 'login', 'wpcomvip' );
```

Comment:
This does the user lookup, which might slow down sites that don't have Akismet as you do this lookup before you check the class exists.

Will it keep trying on every request, or is that scenario valid?

I guess we could load the wpcomvip user only when we need it like few lines bellow

Reply:
I think it's indeed better to do the check after `is_akismet_valid`, since it is only in that case where we need to do the user switch.